src/LanguageServer/Impl/Implementation/RunningDocumentIndexer.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.PythonTools.Analysis;
+using Microsoft.PythonTools.Analysis.Indexing;
+using Microsoft.PythonTools.Parsing;
+using Microsoft.PythonTools.Parsing.Ast;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Python.LanguageServer.Implementation {
+    class RunningDocumentIndexer {
+        private SymbolIndex _index;
+        private PythonLanguageVersion _pythonLanguageVersion;
+        private WorkspaceManager _workspaceManager;
+
+        public RunningDocumentIndexer(SymbolIndex index, PythonLanguageVersion pythonLanguageVersion, WorkspaceManager workspaceManager) {
+            _index = index;
+            _pythonLanguageVersion = pythonLanguageVersion;
+            _workspaceManager = workspaceManager;
+        }
+
+        public void OpenDocument(Uri uri, IDocument doc) {
+            _index.UpdateParseTree(uri, new PythonAst(DocAsts(doc)));
```

Comment:
You prob want to look into my branch, there are no more document parts etc. One AST per file, single part.